programs/bpf_loader/src/syscalls.rs

```diff
@@ -2118,976 +2285,3475 @@ fn call<'a>(
     Ok(SUCCESS)
 }
 
-#[cfg(test)]
-mod tests {
-    use super::*;
-    use solana_rbpf::{
-        ebpf::HOST_ALIGN, memory_region::MemoryRegion, user_error::UserError, vm::Config,
-    };
-    use solana_sdk::{
-        bpf_loader,
-        fee_calculator::FeeCalculator,
-        hash::hashv,
-        process_instruction::{MockComputeMeter, MockInvokeContext, MockLogger},
+/// BIGNUM data cost
+const BIGNUM_WORDCOST: f64 = 6.0;
+/// BIGNUM cost divisor
+const BIGNUM_WORDCOST_DIVISOR: f64 = 32.0;
+/// Bignum data cost calculator
+macro_rules! calc_bignum_cost {
+    ($input:expr) => {
+        (BIGNUM_WORDCOST * ($input / BIGNUM_WORDCOST_DIVISOR).ceil()) as u64
     };
-    use std::str::FromStr;
+}
 
-    const DEFAULT_CONFIG: Config = Config {
-        max_call_depth: 20,
-        stack_frame_size: 4_096,
-        enable_instruction_meter: true,
-        enable_instruction_tracing: false,
-    };
+struct SyscallBigNumFromU32<'a> {
+    cost: u64,
+    compute_meter: Rc<RefCell<dyn ComputeMeter>>,
+    loader_id: &'a Pubkey,
+}
+impl<'a> SyscallObject<BpfError> for SyscallBigNumFromU32<'a> {
+    fn call(
+        &mut self,
+        bn_out_addr: u64,
+        u64_out_len: u64,
+        u32_val: u64,
+        _arg4: u64,
+        _arg5: u64,
+        memory_mapping: &MemoryMapping,
+        result: &mut Result<u64, EbpfError<BpfError>>,
+    ) {
+        question_mark!(
+            self.compute_meter
+                .consume(self.cost + calc_bignum_cost!(u64_out_len as f64)),
+            result
+        );
+        let byte_slice = question_mark!(
+            translate_slice_mut::<u8>(
+                memory_mapping,
+                bn_out_addr,
+                u64_out_len,
+                self.loader_id,
+                true
+            ),
+            result
+        );
+        let bn_bytes = BigNum::from_u32(u32_val as u32).unwrap().to_vec();
```

Comment:
We can't panic in the runtime and even if there is currently no way for this call to fail we try to avoid `unwrap` so that as the code evolves and assumptions change we don't get bit by an `unwrap` panic.

Reply:
While you call out this as an example, I presume it applies to all unwraps I have....?

Reply:
Fixed in a156563ebf54e8934b207cb12cb2d8aefb83b93d